Vendor/iOS/MSAlertController/MSAlertController.m

```diff
@@ -146,18 +147,37 @@ + (void)presentNextAlertAnimated:(BOOL)animated {
   }
 }
 
-// FIXME: This macro is not safe for nil arguments as types in this case cannot be inferred.
+#define ARRAY_FROM_ARGS(...) ({\
+ [NSArray arrayWithObjects: !(sizeof( (char[]){#__VA_ARGS__} ) == 1) ? __VA_ARGS__ : [NSNull null], nil];\
+})
+
+#define MS_DISPATCH_SELECTOR_OBJECT(result, class, selectorName, ...) ({ \
+ NSArray* array = ARRAY_FROM_ARGS(__VA_ARGS__);\
+ id instance = class;\
+ NSInvocation *impl = [MSPerformSelectorUtil invoke:instance withSelector:@#selectorName withObjects:array];\
+ void *results;\
+ [impl getReturnValue:&results];\
+ (__bridge result)results; \
+})
+
 #define MS_DISPATCH_SELECTOR(result, class, selectorName, ...) ({ \
-  SEL selector = NSSelectorFromString(@#selectorName); \
-  IMP impl = [class methodForSelector:selector]; \
-  ((result (*)(id, SEL, ...)) impl)(class, selector, ##__VA_ARGS__); \
+ NSArray* array = ARRAY_FROM_ARGS(__VA_ARGS__);\
+ id instance = class;\
+ NSInvocation *impl = [MSPerformSelectorUtil invoke:instance withSelector:@#selectorName withObjects:array];\
+ void *results = nil;\
+ if(![@#result isEqualToString:@"void"]) {\
+  NSUInteger length = [[impl methodSignature] methodReturnLength];\
+  results = malloc(length);\
+  [impl getReturnValue:&results];\
+ } \
+ (result)results; \
```

Comment:
Consider to rename type to highlight that it's a type

Reply:
Renamed